scheduler/core/components/optimizer/greedymax.py

```diff
@@ -628,7 +628,7 @@ def _charge_time(observation: Observation, atom_start: int = 0, atom_end: int =
             atom_end += seq_length
 
         # Update observation status
-        if atom_end == seq_length:
+        if atom_end == seq_length - 1:
```

Comment:
So this explains why we were getting strange results with observations being marked `READY` despite the fact that they were complete?